runtimes/eden/src/migrations.rs

```diff
@@ -15,23 +15,47 @@ where
-		StorageVersion::new(1).put::<pallet_preimage::Pallet<T>>();
-		// pallet_uniques adding a storage version not chaning anything
+		// pallet_uniques adding a storage version not changing anything
 		StorageVersion::new(1).put::<pallet_uniques::Pallet<T>>();
 
-		// Store version 0 for default
-		StorageVersion::new(0).put::<pallet_nodle_uniques::Pallet<T>>();
+		// Version 1.5 Fixup two unmigrated storage blocks found on mainnet. HostConfiguration is automatically reloaded and Preimage/StatusFor is unused.
+		// [2024-02-06T11:29:36Z ERROR runtime::executive] - 0. error: Failed to decode storage item `ParachainSystem::HostConfiguration`
+		// [2024-02-06T11:29:36Z DEBUG runtime::executive] - 0. error: TryDecodeEntireStorageError { key: [69, 50, 61, 247, 204, 71, 21, 11, 57, 48, 226, 102, 107, 10, 163, 19, 197, 34, 35, 24, 128, 35, 138, 12, 86, 2, 27, 135, 68, 160, 7, 67], raw: Some([0, 0, 48, 0, 0, 80, 0, 0, 170, 170, 2, 0, 0, 0, 16, 0, 251, 255, 0, 0, 16, 0, 0, 0, 10, 0, 0, 0, 64, 56, 0, 0, 88, 2, 0, 0]), info: StorageInfo { pallet_name: [80, 97, 114, 97, 99, 104, 97, 105, 110, 83, 121, 115, 116, 101, 109], storage_name: [72, 111, 115, 116, 67, 111, 110, 102, 105, 103, 117, 114, 97, 116, 105, 111, 110], prefix: [69, 50, 61, 247, 204, 71, 21, 11, 57, 48, 226, 102, 107, 10, 163, 19, 197, 34, 35, 24, 128, 35, 138, 12, 86, 2, 27, 135, 68, 160, 7, 67], max_values: Some(1), max_size: None } }
+		// [2024-02-06T11:29:36Z ERROR runtime::executive] - 1. error: Failed to decode storage item `Preimage::StatusFor`
+		// [2024-02-06T11:29:36Z DEBUG runtime::executive] - 1. error: TryDecodeEntireStorageError { key: [216, 243, 20, 183, 244, 230, 176, 149, 240, 248, 238, 70, 86, 164, 72, 37, 85, 177, 174, 142, 206, 213, 82, 47, 60, 64, 73, 188, 132, 237, 164, 168, 134, 165, 126, 209, 14, 33, 121, 241, 141, 181, 169, 30, 183, 253, 57, 95, 170, 112, 42, 188, 5, 207, 200, 241, 243, 215, 36, 138, 208, 108, 237, 135], raw: Some([0, 1, 74, 34, 11, 19, 82, 60, 166, 200, 236, 155, 46, 110, 132, 247, 154, 76, 85, 37, 199, 188, 137, 30, 55, 23, 255, 89, 202, 125, 127, 172, 65, 44, 0, 67, 52, 105, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), info: StorageInfo { pallet_name: [80, 114, 101, 105, 109, 97, 103, 101], storage_name: [83, 116, 97, 116, 117, 115, 70, 111, 114], prefix: [216, 243, 20, 183, 244, 230, 176, 149, 240, 248, 238, 70, 86, 164, 72, 37, 85, 177, 174, 142, 206, 213, 82, 47, 60, 64, 73, 188, 132, 237, 164, 168], max_values: None, max_size: None } }
 
-		T::DbWeight::get().writes(6)
+		//Host Configuration
+		// TODO Remove this migration at final merge. It is only needed during integration (as it will fail one test in CI and hide other real errors)
+		let parachain_system_host_configuration_key = [
+			69_u8, 50, 61, 247, 204, 71, 21, 11, 57, 48, 226, 102, 107, 10, 163, 19, 197, 34, 35, 24, 128, 35, 138, 12,
+			86, 2, 27, 135, 68, 160, 7, 67,
+		];
+		let _ = frame_support::storage::unhashed::clear_prefix(&parachain_system_host_configuration_key, Some(1), None);
```

Comment:
I know we routinely remove the migration logic and thus it's expected to be executed only once. However, do you think it would be safer to guard this with a check just in case this logic remains here longer than it should, it doesn't do any harm? A check that shows this is not done before could be worth adding here.

Reply:
It is copied back after runtime-upgrade, so there was no harm, we could remove it but then the CI tests will fail(and potentially hide real problems) until we upgrade the chain.